### ISSUE TITLE
PG::InsufficientPrivilege: ERROR:  permission denied to create extension "hstore"

### ISSUE THREAD

**Issue by zenati** [dokku-alt/dokku-alt#66]
I have a Rails application and a migration that runs :

```
execute("CREATE EXTENSION IF NOT EXISTS hstore")
```

But after creating a postgresql database using :

```
dokku postgresql:create appdb
```

and linking it to the app using :

```
dokku postgresql:link app appdb
```

I run:

```
dokku run app rake db:migrate
```

And I get this error :

```
Migrating to CreateMetaTagsList (20140919122507)
== 20140919122507 CreateMetaTagsList: migrating ===============================
-- create_table(:meta_tags_lists)
   -> 0.0067s
== 20140919122507 CreateMetaTagsList: migrated (0.0072s) ======================

Migrating to CreateParaComponents (20140926102843)
== 20140926102843 CreateParaComponents: migrating =============================
-- execute("CREATE EXTENSION IF NOT EXISTS hstore")
PG::InsufficientPrivilege: ERROR:  permission denied to create extension "hstore"
HINT:  Must be superuser to create this extension.
: CREATE EXTENSION IF NOT EXISTS hstore
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:

PG::InsufficientPrivilege: ERROR:  permission denied to create extension "hstore"
HINT:  Must be superuser to create this extension.
```

Any idea on why I can't create that extension ?

**Comment by scifisamurai** [comment]
Thanks @cadicallegari I was running into a permissions issue with postgresql not specific to docker and your solution: sudo su postgres -c "psql db_name -c 'CREATE EXTENSION hstore;'" worked! Thanks again :smiley:

**Comment by henrik** [comment]
@cadicallegari Thanks so much for this! I had similar trouble with dokku (non-alt).

In an attempt to clarify your solution further for others:

* SSH to your dokku(-alt) server
* Run `docker ps` to see the container ID of your postgres
* Run `dokku config my-app` to see the database name of your app (e.g. `db_my_app`)
* Run that command, substituting those values

I actually got an error like `System error: exec: "sudo": executable file not found in $PATH`. I guess my postgres plugin (again, not in -alt) does not come with "sudo". I just removed the "sudo", keeping everything else, and that worked.
